esmvaltool/_data_finder.py

```diff
@@ -362,67 +380,89 @@ def get_input_filelist(variable, rootpath, drs):
                         valid_dirs.append(dirname)
                         break
             else:
-                raise IOError(
-                    'Path {} does not exist'.format(part1))
+                logger.debug('Path %s does not exist', part1)
 
     # Set the filename glob
     filename_glob = _get_filename(variable, drs)
 
-    for dir_name in valid_dirs:
-        # Find files
-        files = find_files(dir_name, filename_glob)
+    if valid_dirs:
+        for dir_name in valid_dirs:
+            # Find files
+            files = find_files(dir_name, filename_glob)
 
-        # Select files within the required time interval
-        files = select_files(files, variable['start_year'],
-                             variable['end_year'])
-        all_files.extend(files)
+            # Select files within the required time interval
+            files = select_files(files, variable['start_year'],
+                                 variable['end_year'])
+            all_files.extend(files)
+    else:
+        all_files = []
 
     return all_files
 
 
 def get_input_fx_filelist(variable, rootpath, drs):
     """Return the full path to input files."""
     dirname_templates = get_input_fx_dirname_template(variable, rootpath, drs)
-    fx_files = {}
+    dirnames = []
 
-    for j, dirname_template in zip(
-            range(len(dirname_templates)), dirname_templates):
+    for dirname_template in dirname_templates:
         # Find latest version if required
         if '[latestversion]' in dirname_template:
             part1, part2 = dirname_template.split('[latestversion]')
             part2 = part2.lstrip(os.sep)
-            # root part1 could not exist at all
-            if not os.path.exists(part1):
-                fx_files[variable['fx_files'][j]] = None
-                return fx_files
-            list_versions = os.listdir(part1)
-            list_versions.sort(reverse=True)
-            if 'latest' in list_versions:
-                list_versions.insert(
-                    0, list_versions.pop(list_versions.index('latest')))
-            for version in list_versions:
-                if version == 'latest':
-                    dirname = os.path.join(part1, version, part2)
-                    if os.path.isdir(dirname):
-                        break
-                else:
-                    dirname = os.path.join(part1, version, part2)
-                    if os.path.isdir(dirname):
-                        break
+
+            # if part1 exists, check versioning
+            if os.path.exists(part1):
+                list_versions = os.listdir(part1)
+                list_versions.sort(reverse=True)
+                if 'latest' in list_versions:
```

Comment:
I think this block of code [line 418 - 432] can be rewritten as:
```python
for version in ['latest'] + list_versions:
    dirname = os.path.join(part1, version, part2)
    if os.path.isdir(dirname):
       dirnames.append(dirname)
       break
```

Reply:
The code
```
if 'latest' in list_versions:
    list_versions.insert(0, list_versions.pop(list_versions.index('latest')))
```
can be removed now.

Reply:
done! cheers for the catch, it was trying to escape :grin: